frontend/model/database.js

```diff
@@ -13,39 +13,38 @@ const log = localforage.createInstance({
 const isLightweightClient = process.env.LIGHTWEIGHT_CLIENT === 'true'
 
 // make gi.log use localforage for storage
-sbp('sbp/selectors/overwrite',
-  isLightweightClient
-    ? {
-        'gi.db/get': key => {
-          const contractId = sbp('gi.db/log/contractIdFromLogHEAD', key)
-          if (!contractId) {
-            return null
-          }
-          const state = sbp('state/vuex/state')
-          if (state.contracts[contractId]?.HEAD) {
-            return state.contracts[contractId].HEAD
-          }
+sbp('sbp/selectors/overwrite', isLightweightClient
+  ? {
+      'gi.db/get': key => {
+        const contractId = sbp('gi.db/log/contractIdFromLogHEAD', key)
+        if (!contractId) {
           return null
-        },
-        'gi.db/set': (key, value) => {
-          return Promise.resolve(value)
-        },
-        'gi.db/delete': () => {
-          return Promise.resolve()
         }
-      }
-    : {
-        'gi.db/get': key => {
-          return log.getItem(key)
-        },
-        // TODO: handle QuotaExceededError
-        'gi.db/set': (key, value) => {
-          return log.setItem(key, value)
-        },
-        'gi.db/delete': (key: string) => {
-          return log.removeItem(key)
+        const state = sbp('state/vuex/state')
+        if (state.contracts[contractId]?.HEAD) {
+          return state.contracts[contractId].HEAD
         }
-      })
+        return null
+      },
+      'gi.db/set': (key, value) => {
+        return Promise.resolve(value)
+      },
+      'gi.db/delete': () => {
+        return Promise.resolve()
+      }
+    }
+  : {
+      'gi.db/get': key => {
+        return log.getItem(key)
+      },
+      // TODO: handle QuotaExceededError
+      'gi.db/set': (key, value) => {
+        return log.setItem(key, value)
+      },
+      'gi.db/delete': (key: string) => {
+        return log.removeItem(key)
+      }
+    })
```

Comment:
Great job. We should also lock the selectors after overwriting them so that they cannot be overwritten again:

```js
sbp('sbp/selectors/lock', ['gi.db/get', 'gi.db/set', 'gi.db/delete'])
```